pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -213,53 +216,13 @@ def work(self):
                                         else:
                                             logger.log(
                                             'Failed to evolve {}!'.format(pokemon_name))
-
-                                    if self.should_release_pokemon(pokemon_name, cp, pokemon_potential, response_dict):
-                                        # Transfering Pokemon
-                                        pokemon_to_transfer = list(
-                                            Set(id_list2) - Set(id_list1))
-                                        if len(pokemon_to_transfer) == 0:
-                                            raise RuntimeError(
-                                                'Trying to transfer 0 pokemons!')
-                                        self.transfer_pokemon(pokemon_to_transfer[0])
-                                        self.bot.metrics.released_pokemon()
-                                        logger.log(
-                                            '{} has been exchanged for candy!'.format(pokemon_name), 'green')
+                                            
+                                    worker = PokemonTransferWorker(self.bot)
+                                    if worker.should_release_pokemon(pokemon_name, cp, pokemon_potential):
+                                        worker.release_catched_pokemon(pokemon_to_transfer[0])
```

Comment:
@surfaace this shouldn't be here at all. This worker is already instantiated in the event loop at every tick. You don't need to call it here, it will be called after every small step the bot takes.